nntrainer/layers/operation_layer.h

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 SeungBaek Hong <sb92.hong@samsung.com>
+ *
+ * @file   operation_layer.h
+ * @date   4 Oct 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author SeungBaek Hong <sb92.hong@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is common class for operation layers
+ *
+ */
+#ifndef __LAYER_OPERATION_H__
+#define __LAYER_OPERATION_H__
+#ifdef __cplusplus
+
+#include <layer_context.h>
+#include <layer_devel.h>
+
+namespace nntrainer {
+
+/**
+ * @brief Type of layer depending on the number of inputs.
+ *
+ */
+enum class OperationType { NONE, UNARY, BINARY };
```

Comment:
It is important to consider whether the operation can be performed in-place to save memory. I guess it might be good to give the user an option to set it when the operation layer is created.

Reply:
A new pull request will be uploaded concerning inplace operations. This implementation has been updated based on the UnaryOperationLayer and BinaryOperationLayer classes! Thanks